src/clj/qbits/jet/client/http.clj

```diff
@@ -281,15 +282,29 @@
     (.setAccessible true)
     (.set request query-string)))
 
+(defn- wrap-iterator
+  [on-last-chunk ^Iterator iterator]
+  (reify
+    Iterator
+    (hasNext [_]
+      (let [more? (.hasNext iterator)]
+        (when-not more?
+          (on-last-chunk))
+        more?))
+    (next [_]
+      (.next iterator))))
```

Comment:
Does this function need the `on-last-chunk` logic as well?  Or at least a comment saying that `hasNext` must always be called before `next`?

Reply:
Added comments clarifying the contract for use.